### ISSUE TITLE
Suggestion: use global (/local) config file for default settings

### ISSUE THREAD

**Issue by aldanor** [joeyespo/pytest-watch#20]
Just an idea.

I found myself using the same set of options over and over, e.g. like so:

```sh
$ ptw --nobeep --poll --ignore=.tox -- \
    --import . tests -sv \
    --cov=some.package --cov-report=term \
    --cov-report=html --cov-config=tox.ini
```

Maybe we could support loading the defaults from `~/.ptwrc` which could look like this
```ini
[pytest-watch]
nobeep = true
poll = true
ignore = .tox
addopts = --import . --cov-report=term --cov-report=html --cov-config=tox.ini -sv
```

and then the command above would be shortened to just

```sh
$ ptw -- --cov=some.package  # default options, with coverage
$ ptw                        # default options, no coverage
```

Maybe it could also look for a local `./.ptwrc` file first (in the current folder).

**Comment by joeyespo** [comment]
Makes sense to me.

Are you interested in implementing?

For reference, here's how pytest [searches for the config file](https://pytest.org/latest/customize.html). I think we could even `import pytest` and use its API to find the config file so we're not duplicating any code.

Thanks for the suggestion!

**Comment by aldanor** [comment]
Hmm, instead of using a different config file we could just store it as a section in one of `tox.ini`, `pytest.ini` and just use pytest machinery to pick it up. Many packages already do that (e.g. `flake8`, `pep8`, `coverage` support all of those standard files).

I'm interested in implementing but realistically I won't have any time for that till a few weeks from now :/

**Comment by joeyespo** [comment]
> instead of using a different config file we could just store it as a section in one of tox.ini, pytest.ini and just use pytest machinery to pick it up

Oh yes, that's what I was suggesting. (Guess I wasn't too clear about re-using `pytest.ini` though.)

> I'm interested in implementing but realistically I won't have any time for that till a few weeks from now :/ 

It's all good! I'll mark this as "help wanted" for now. Thanks!

**Comment by joeyespo** [comment]
Just published v3.8.0 which contains this feature.

The implementation doesn't contain `addopts` though because you can add config directly to the `[pytest]` section instead. Let me know if this help, @aldanor.

Thanks, @aldanor and @bendtherules!

**Comment by aldanor** [comment]
:+1:
